File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
//1260
#include <iostream>
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;

vector<vector<int>> a;
vector<bool> visited;

void dfs(int x) {
    visited[x] = true;
    cout << x << " ";

    for (int i = 0; i < a[x].size(); i++) {
        int k = a[x][i];
        if (!visited[k]) {
            dfs(k);
        }
    }
}

void bfs(int start) {
    queue<int> q;
    visited[start] = true;
    q.push(start);

    while (!q.empty()) {
        int x = q.front();
        q.pop();
        cout << x << " ";

        for (int i = 0; i < a[x].size(); i++) {
            int k = a[x][i];
            if (!visited[k]) {
                visited[k] = true;
                q.push(k);
            }
        }
    }
}

int main() {
    int n, m, start;
    cin >> n >> m >> start; // n: node, m: edge, start: 시작 노드

    a.resize(n + 1); // 노드 번호가 1부터 시작하므로 n+1 크기의 벡터 초기화
    visited.resize(n + 1, false); // 방문 여부 배열 초기화

    for (int i = 0; i < m; i++) {
        int u, v;
        cin >> u >> v;

        a[u].push_back(v);
        a[v].push_back(u);
    }

    for (int i = 1; i <= n; i++) {
        sort(a[i].begin(), a[i].end()); // 방문할 노드 순서대로 정렬
    }

    dfs(start);
    cout << endl;

    fill(visited.begin(), visited.end(), false); //방문 다시 초기화
    bfs(start);
    cout << endl;

    return 0;
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
#1260
import sys
from collections import deque
    
def input():
    return sys.stdin.readline().rstrip()

def dfs(graph, v, visited):
    visited[v] = True
    print(v, end=' ')
    
    for neighbor in sorted(graph[v]):
        if not visited[neighbor]:
            dfs(graph, neighbor, visited)
    
def bfs(graph, v, visited):
    queue = deque([start])
    visited[start] = True
    
    while queue:
        v = queue.popleft()
        print(v, end=' ')
        for neighbor in sorted(graph[v]):
            if not visited[neighbor]:
                queue.append(neighbor)
                visited[neighbor] = True

node, edge, start = map(int, input().split())
graph = [[] for _ in range(node+1)]

for _ in range(edge):
    u, v = map(int, input().split())
    graph[u].append(v)
    graph[v].append(u)

# dfs
visited = [False] * (node + 1)
dfs(graph, start, visited)
print()

# bfs
visited = [False] * (node + 1)
bfs(graph, start, visited)